### ISSUE TITLE
Polybar doesn't recognize VGA monitor width, causes error to launch

### ISSUE THREAD

**Issue by ilzichio** [polybar/polybar#1794]
## Describe the issue
`polybar example` doesn't work because polybar doesn't recognize exteranal monitor width 

**Expected behavior:**

A working polybar

**Actual behavior:**

`polybar -m` output: 

```
VGA1: 0x900+0+0
LVDS1: 1600x900+0+0 (primary)

```
Width for some reason is zero

**Was it working before?**

No, I just installed it from AUR

## To Reproduce
polybar config
```
[bar/example]
monitor = VGA1
width = 5%
height = 27
;offset-x = 1%
;offset-y = 1%
radius = 6.0
fixed-center = false


```

## Polybar Log

```
$ polybar example
error: Resulting bar width is out of bounds (0)

```

## Environment:
* WM: i3-gaps
* Distro: Arch Linux
* Output of `polybar -vvv`: polybar 3.3.1

```
Features: +alsa +curl +i3 -mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync +composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

XRandr output:

```
$ xrandr -q | grep -w connected
LVDS1 connected primary 1600x900+0+0 (normal left inverted right x axis y axis) 310mm x 170mm
VGA1 connected 1440x900+0+0 (normal left inverted right x axis y axis) 410mm x 260mm   

$ xrandr --listmonitors
Monitors: 2
 0: +*LVDS1 1600/310x900/170+0+0  LVDS1
 1: +VGA1 1440/410x900/260+0+0  VGA1

```

**Comment by patrick96** [comment]
This is likely caused by the same issue as #1437. Polybar doesn't properly handle the new xrandr "monitors" in some circumstances.

One thing you can try to fix this right now is to compile without randr monitor support.
You do this by adding `-DWITH_XRANDR_MONITORS=OFF` as a command line argument to cmake during compilation.

Since you are installing from the AUR you can modify the second line (line 34 probably) of the `polybar` `PKGBUILD` to read:

```bash
  cmake -DCMAKE_INSTALL_PREFIX=/usr -DWITH_XRANDR_MONITORS=OFF ..
```

Please report back whether or not this worked.

**Comment by patrick96** [comment]
I just got another idea. This may be related to #1191 since your VGA monitor occupies the same space as the LVDS1 monitor and we had issues with that before. Would you mind trying to use the fix in #1823 and see if this fixes it.

---
Technical Note:

This would mean the process that removes cloned monitors from the output of `polybar -m` is also bugged. I suspect it's because the clone removal code looks like this:
```
    for (monitor m in monitors) {
      if ((*m)->w == 0) {
        remove_monitor(*m);
        continue;
      }

      // Test if there are any clones in the set
      for (auto& monitor : monitors) {
        if (monitor is completely contained in m) {
          // Reset width so that the output gets
          // removed in the base loop
          monitor->w = 0;
        }
      }
    }
```

If `monitors = ['VGA', 'LVDS1']`, in the second iteration of the outer loop we are on `LVDS1` and in the inner loop we detect that `VGA` is contained in `LVDS1` and set the width to 0 expecting it will get removed in the outer loop. But we already looped over VGA so it's not removed but its width set to 0. I think I can fix this in #1823 as well.

**Comment by patrick96** [comment]
I was able to reproduce this as well now on my machine and #1823 should fix it.
